SportsHub/SportsHub.Api/Controllers/CommentController.cs

```diff
@@ -1,33 +1,93 @@
 ﻿using AutoMapper;
+using FluentValidation;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using SportsHub.Api.Validations;
+using SportsHub.AppService.Authentication.Models.DTOs;
 using SportsHub.AppService.Services;
+using SportsHub.Domain.Constants;
+using SportsHub.Domain.Models;
 
 namespace SportsHub.Api.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class CommentController : ControllerBase
     {
-        private readonly ICommentService _service;
+        private readonly ICommentService _commentService;
         private readonly IMapper _mapper;
+        private readonly IValidator<CreateCommentDTO> _commentValidator;
+        private readonly IGenerateModelStateDictionary _generateModelStateDictionary;
 
-        public CommentController(ICommentService service, IMapper mapper)
+        public CommentController(ICommentService service, IMapper mapper, IValidator<CreateCommentDTO> commentValidator, IGenerateModelStateDictionary generateModelStateDictionary)
```

Comment:
You use model from application layer as request model. It is better not to do so. Request model should be declared on API layer, and it should be mapped to domain model or model from service layer in action.

Reply:
Implemented.